tests/queries_test.go

```diff
@@ -21,11 +29,54 @@ func TestInternal(t *testing.T) {
 }
 
 func TestCommands(t *testing.T) {
+	systest(t)
+	path := findPath()
 	for _, cmd := range commands {
 		t.Run(cmd.Name, func(t *testing.T) {
-			results, err := cmd.Run()
+			results, err := cmd.Run(path)
 			require.NoError(t, err)
 			assert.Exactly(t, cmd.Expected, results, "Wrong command results")
 		})
 	}
 }
+
+func findPath() string {
+	for _, s := range os.Args {
+		if strings.HasPrefix(s, "PATH=") {
+			return s[5:]
+		}
+	}
+	return ""
+}
+
+func TestScripts(t *testing.T) {
+	systest(t)
+	path := findPath()
+	for _, script := range scripts {
+		t.Run(script.Name, func(t *testing.T) {
+			var fail bool
+			shell := test.NewShellTest(script)
+			_, _, err := shell.Run(RootDir, path)
+			if err != nil {
+				fail = true
+			}
+			require.NoError(t, err)
+			for _, file := range script.Expected {
+				actual, err := shell.Read(file.Name)
+				if err != nil {
+					fail = true
+				}
+				require.NoError(t, err)
+				if !assert.Exactly(t, file.Data, actual, "Wrong shell script results") {
+					fail = true
+				}
+			}
+			if !fail {
+				// Remove the testdir on success.  If test fails,  we
+				// leave it behind in testroot for debugging.  These
+				// failed test directories have to be manually removed.
+				shell.Cleanup()
+			}
```

Comment:
Feels like root-dir should be cleared on the next run... my started getting a lot of gunk in it.

Reply:
I kind of like the idea but it doesn't feel like expected behavior and could cause surprises.